backend/lib/services/terminals/terminalBootstrap.js

```diff
@@ -100,28 +106,33 @@ class BootstrapMap extends Map {
     this.set(key, value)
   }
 
-  setInProgress (key) {
+  setInProgress (item) {
+    const key = this.getKey(item)
     const value = this.getValue(key)
     value.state = BootstrapStatusEnum.IN_PROGRESS
     this.set(key, value)
   }
 
-  setPending (key) {
+  setPostponed (item) {
+    const key = this.getKey(item)
     const value = this.getValue(key)
-    value.state = BootstrapStatusEnum.PENDING
+    value.state = BootstrapStatusEnum.POSTPONED
     this.set(key, value)
   }
 
-  getValueForResource (resource) {
-    const key = this.getKeyForResource(resource)
-    return this.getValue(key)
-  }
-
-  getValue (key) {
+  getValue (item) {
+    const key = this.getKey(item)
     return this.get(key) || {
       state: BootstrapStatusEnum.INITIAL
     }
   }
+
```

Comment:
The class `BootstrapMap` has only one subclass. Move the methods of the single subclass into the `BootstrapMap` class. Merge `getKey` and `getKeyForResource` into one implementation.

Reply:
✅